### DISCUSSION TITLE
`/dev/serial/by-id/` path doesn't exists

### DISCUSSION THREAD

**Discussion by TonySeeley** [zwave-js/zwave-js-ui#3589] [Q&A] [answered]
Hi I've set up zwave-js-ui and am having trouble with zwave seeing my device (usb dongle).
In the documentation it tells you NOT to use /dev/ttyxxxx but to use the full /dev/serial/by-id/* device identifier, but in my version of ubuntu server, running a "ls -l /dev/serial/by-id" command doesnt work anymore, they've removed the "by-id" symlinks.
How can I configure the device link between the host and the zwave-js-ui container.
Obviously I've tried the : 
    devices:
      - /dev/ttyACM0:/dev/zwave'
 and as stated in the docs, it doesn't work.
Any help would be Really appreciated.
Thanks

**Comment by robertsLando** [comment, marked as answer]
Tried to also check `ls /dev/serial/by-path/` ? Otherwise try to check this: https://www.reddit.com/r/debian/comments/1331wlr/devserialbyid_suddenly_missing/?rdt=45983

TLDR;

```
sudo nano /lib/udev/rules.d/60-serial.rules
```

Create a backup of the content of that file and then replace it with this:

```
# do not edit this file, it will be overwritten on update

ACTION=="remove", GOTO="serial_end"
SUBSYSTEM!="tty", GOTO="serial_end"

SUBSYSTEMS=="usb", IMPORT{builtin}="usb_id", IMPORT{builtin}="hwdb --subsystem=usb"
SUBSYSTEMS=="pci", ENV{ID_BUS}=="", ENV{ID_BUS}="pci", \
  ENV{ID_VENDOR_ID}="$attr{vendor}", ENV{ID_MODEL_ID}="$attr{device}", \
  IMPORT{builtin}="hwdb --subsystem=pci"

# /dev/serial/by-path/, /dev/serial/by-id/ for USB devices
KERNEL!="ttyUSB[0-9]*|ttyACM[0-9]*", GOTO="serial_end"

SUBSYSTEMS=="usb-serial", ENV{.ID_PORT}="$attr{port_number}"

IMPORT{builtin}="path_id"
ENV{ID_PATH}=="?*",                   ENV{.ID_PORT}=="",   SYMLINK+="serial/by-path/$env{ID_PATH}"
ENV{ID_PATH_WITH_USB_REVISION}=="?*", ENV{.ID_PORT}=="",   SYMLINK+="serial/by-path/$env{ID_PATH_WITH_USB_REVISION}"
ENV{ID_PATH}=="?*",                   ENV{.ID_PORT}=="?*", SYMLINK+="serial/by-path/$env{ID_PATH}-port$env{.ID_PORT}"
ENV{ID_PATH_WITH_USB_REVISION}=="?*", ENV{.ID_PORT}=="?*", SYMLINK+="serial/by-path/$env{ID_PATH_WITH_USB_REVISION}-port$env{.ID_PORT}"

ENV{ID_BUS}=="", GOTO="serial_end"
ENV{ID_SERIAL}=="", GOTO="serial_end"
ENV{ID_USB_INTERFACE_NUM}=="", GOTO="serial_end"
ENV{.ID_PORT}=="", SYMLINK+="serial/by-id/$env{ID_BUS}-$env{ID_SERIAL}-if$env{ID_USB_INTERFACE_NUM}"
ENV{.ID_PORT}=="?*", SYMLINK+="serial/by-id/$env{ID_BUS}-$env{ID_SERIAL}-if$env{ID_USB_INTERFACE_NUM}-port$env{.ID_PORT}"

LABEL="serial_end"
```

Then relopad rules:

```
sudo udevadm trigger
```

Check if now path exists otherwise try to unplug and plug back the stick and check again, if still not present try a reboot:

```
sudo reboot
```

And check again

**Reply by TonySeeley** [reply]
Wow thank you, this worked brilliantly.
